### ISSUE TITLE
Getting all contracts for the symbol

### ISSUE THREAD

**Issue by levshinD** [Voyz/ibind#37]
I'm trying to search for all available contracts for a symbol, cant figure out how to do it with your API

**Comment by Voyz** [comment]
See one of these:

- https://github.com/Voyz/ibind/wiki/API-Reference-%E2%80%90-IbkrClient#client.ibkr_client_mixins.contract_mixin.ContractMixin.all_conids_by_exchange
- https://github.com/Voyz/ibind/wiki/API-Reference-%E2%80%90-IbkrClient#client.ibkr_client_mixins.contract_mixin.ContractMixin.security_stocks_by_symbol
- https://github.com/Voyz/ibind/wiki/API-Reference-%E2%80%90-IbkrClient#client.ibkr_client_mixins.contract_mixin.ContractMixin.search_contract_by_symbol

**Comment by levshinD** [comment]
https://github.com/Voyz/ibind/wiki/API-Reference-%E2%80%90-IbkrClient#client.ibkr_client_mixins.contract_mixin.ContractMixin.search_contract_by_symbol

using this call and get only one contract and list of months

**Comment by Voyz** [comment]
How about this one?

https://github.com/Voyz/ibind/wiki/API-Reference-%E2%80%90-IbkrClient#client.ibkr_client_mixins.contract_mixin.ContractMixin.security_stocks_by_symbol

**Comment by levshinD** [comment]
do you mind sharing a code example, i keep getting exceptions. I would like to get all the contracts for IBIT for all durations and all strikes

**Comment by levshinD** [comment]
found an example in stock querying

**Comment by levshinD** [comment]
for a stock, but not for an option

**Comment by levshinD** [comment]
{"IBIT":[{"name":"ISHARES BITCOIN TRUST ETF","chineseName":null,"assetClass":"STK","contracts":[{"conid":677037673,"exchange":"NASDAQ","isUS":true}]}]}

i get back only one contract from all the queries, not the chain

**Comment by levshinD** [comment]
i suspect that this call has a bug[
(https://localhost:5000/v1/api/iserver/secdef/info?conid=677037673&sectype=OPT&month=D&strike=15.0&right=C)
the month is not properly sent

**Comment by Voyz** [comment]
> i suspect that this call has a bug[(https://localhost:5000/v1/api/iserver/secdef/info?conid=677037673&sectype=OPT&month=D&strike=15.0&right=C) the month is not properly sent

That's right, the example contained an error. I've fixed it now on the master branch, feel free to try it out.

> do you mind sharing a code example

Just to clarify on this - the rest_06 example was contributed by one of the users as I don't trade options, hence my help in this space will be somewhat limited. But please share what exceptions you're running into and I will try to aid you

**Comment by Voyz** [comment]
I'm going to close this issue due to inactivity. Thanks for your contribution and please feel free to request a reopen if you'd like to continue the discussion 👍
